### ISSUE TITLE
wgsl: clarify that type inference occurs independent of the numeric value of a constant-time expression

### ISSUE THREAD

**Issue by dneto0** [gpuweb/gpuweb#2830]
The conversion rank section says:

>  When a type assertion e:T is used as a type rule precondition, it is satisfied when:

> *  e is already of type T, or

> * the value of e can be automatically converted to a value of type T.

That doesn't correctly capture the decision we had made.

For example:

        let a = 12345;  // infers a : i32

Changing the value to a value that fits in u32 but not i32 should not change the inferred type:

        let a = 4000000000; // should still infer a:i32, 
            // and should lead to shader-creation error because the value is too big.

In particular, this should not infer `a:u32`.

In summary, type inference should be performed independent of the actual numeric value of the constant-time expression

Other relevant examples:

         let a: u32 = 4000000000; //should infer a:u32, and store teh value 4 billion.
         let b = u32(4000000000);  // works, as invocation of u32(AbstractInt) conversion.

**Comment by alan-baker** [comment]
Maybe just a note.

**Comment by dneto0** [comment]
Push to Milestone 1, to add this as a note,. mabye
